component/src/main/java/org/wso2/extension/siddhi/map/text/sinkmapper/TextSinkMapper.java

```diff
@@ -134,15 +136,26 @@
     private String streamID;
 
     @Override
-    public void init(StreamDefinition streamDefinition, OptionHolder optionHolder, TemplateBuilder
-            templateBuilder, ConfigReader configReader, SiddhiAppContext siddhiAppContext) {
+    public void init(StreamDefinition streamDefinition, OptionHolder optionHolder, Map<String,
+            TemplateBuilder> payloadTemplateBuilderMap, ConfigReader mapperConfigReader,
+                     SiddhiAppContext siddhiAppContext) {
         this.streamID = streamDefinition.getId();
         this.attributeList = streamDefinition.getAttributeList();
         this.eventGroupEnabled = Boolean.valueOf(optionHolder
                 .validateAndGetStaticValue(OPTION_GROUP_EVENTS, DEFAULT_GROUP_EVENTS));
         this.endOfLine = optionHolder.validateAndGetStaticValue(OPTION_NEW_LINE, DEFAULT_NEW_LINE);
         this.eventDelimiter = optionHolder.validateAndGetStaticValue(OPTION_GROUP_EVENTS_DELIMITER,
                 DEFAULT_EVENTS_DELIMITER) + endOfLine;
+
+        if (payloadTemplateBuilderMap != null && payloadTemplateBuilderMap.size() != 1) {
```

Comment:
Can't payloadTemplateBuilderMap.size() = 0? If not, can you please add a small comment mentioning why...  Otherwise,  by just looking at the code, it seems you have missed that case...

Reply:
It can't be 0. Update the PR.